COMMANDS.org

```diff
@@ -0,0 +1,117 @@
+- [[#-temporary-packages][Temporary Packages]]
+  - Excellent for testing new versions of packages, or entirely new tools, without comitting to system level changes that could impact other parts of the system.
+
+** 🔍 Searching for Pacakges
+
+#+begin_quote
+/Question: I want to install pacakges, but I'm not sure what is avalible/
+#+end_quote
+
+Search for packages via:
+- The web UI @ https://search.nixos.org/
+- Using the cli:
+  #+begin_src bash
+  nix-env -qaP 'some_package' 
+  #+end_src
+
+  - ~-q~: Stands for "query". Used to query the Nix store for /installed/ pacakges
+  - ~-a~: Stands for "avalible". Used to search for /avalible packages/ to be installed
+  - ~-P~: Includes the attribute path of the package. This result is often used with the ~-A~ flag during installs to more precisely target the desired package.
+
+** 🪛 Imperative Package Management
+
+#+begin_quote
+/Question: I'm used to tools like brew - how do I use nix in a similar way?/
+#+end_quote
+
+If your workflow has involved...
+- ~brew install some_package~
+- ~port install some_package~
+- ~api-get install some_package~
+
+...and are making the move to ~nix~, then you are in the right place! ~nix~ can be used as a drop in replacement for these tools on most linux/unix based systems.
+
+*** Install a package
+
+Once the desired package has been identified, use ~nix-env~ to install the package globally...
+
+#+begin_src bash
+  nix-env -i some_package
+#+end_src
+
+- ~-i~: short flag for ~--install~.
+  - Takes multiple arguements in a space-seperated list.
+  - By default, ~nix-env -i~ will do a lookup of the argument given against ~nixpkgs~, similarly to going to https://search.nixos.org/, searching, and using the top result.
+- ~-A~: short for ~--attr~ across many ~nix~ commands.
+  - Instead of relying on the lookup, which could be slow at times, one could add ~-A~ along with the /exact/ package name to skip the lookup completely. This requires the name given to be exact, and can allow for installing a specific version of the target package.
+  - Can be combined with other flags i.e. ~-iA~
+
+#+begin_quote
+Note: If a package being installed complains about the package license being "unfree", this means the package has some licensing agreement that is not conguent with typical open-source tools.
+
+To install these packages anyway, prepend ~NIXPKGS_ALLOW_UNFREE=1~ to the call to make a 1 time allowance. The whole call should look something like:
+#+begin_src bash
+  NIXPKGS_ALLOW_UNFREE=1 nix-env -i some_package
+#+end_src
+#+end_quote
+
+*** Uninstalling a package
+
+Uninstalling a package requires knowing its name. We can find a list of the installed packages like this:
+
+#+begin_src bash
+  nix-env -q some_package
+#+end_src
+
+With the target package name in hand, we can then run:
+
+#+begin_src bash
+  nix-env -e some_package
+#+end_src
+
+- ~-e~: Stands for "erase"
+  - Takes a space-seperated list of pacakges to be removed from the system
```

Comment:
Similar to earlier, I might just embed `--erase` instead of `-e` here. Engineers that don't want to remember the long name arg know how to use a man page

Well, for that matter, so do the ones that seem to be resisting `nix`, so ¯\\_(ツ)_/¯

-----

Similar to earlier, it might be worth making a subsection for a suggestion like:

```
**** Aliases for simplification
Similar to earlier, you can make a couple of aliases to help you if the brew-isms are easier to remember

#+begin_src bash
alias nix-list = "nix-env -q"
alias nix-remove = "nix-env --erase"
#+end_src
```

Reply:
Yeah - I was of two minds here... On one hand, if you know how to make aliases, you don't need the reminder to - and probably have opinions on how to name them. On the other, if you don't know how to make aliases, would you even know where to put the definitions and then remember they existed when things don't work as expected?

I do like, if they need to be included, having a suggested aliases section instead of sprinkling them all over the document.

Reply:
I can see what you mean, but in context of this doc a link to "what are aliases" might be fine for somebody that needs it